airflow/models.py

```diff
@@ -1369,9 +1369,9 @@ def append_only_new(self, l, item):
     def _set_relatives(self, task_or_task_list, upstream=False):
         if isinstance(task_or_task_list, BaseOperator):
             task_or_task_list = [task_or_task_list]
+        if not isinstance(task_or_task_list, list):
+            raise AirflowException('Expecting a task')
         for task in task_or_task_list:
-            if not isinstance(task_or_task_list, list):
```

Comment:
I think we should test inside the loop if `isinstance(task, BaseOperator)`, since we don't really care that this is a list.
@mistercrunch?